### ISSUE TITLE
Function with result and bind throw a syntax error

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#214]
```fortran 
module mod1
use, intrinsic :: iso_c_binding
contains

  function fct1() result(res) bind(C, name='c_fct1')
    integer :: res
    res = 1
  end function

end module mod1
```

```bash
"../test/testdata/iso_c_binding.f90:mod1", line 5: syntax error
```

This is an easy fix to do in the grammar. I'll do a PR today for this.

**Comment by clementval** [comment]
PR waiting for merge #215

**Comment by clementval** [comment]
Closed with #215
